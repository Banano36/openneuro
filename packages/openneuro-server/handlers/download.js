import { graphql } from 'graphql'
import schema from '../graphql/schema.js'

const DRAFT_FILES = `
  query dataset($datasetId: ID!) {
    dataset(id: $datasetId) {
      id
      draft {
        id
        files {
          id
          filename
          size
          urls
        }
      }
    }
  }
`

const SNAPSHOT_FILES = `
  query snapshot($datasetId: ID!, $tag: String!) {
    snapshot(id: $datasetId) {
      id
      files {
        id
        filename
        size
        urls
      }
    }
  }
`

export const datasetDownload = (req, res) => {
  const datasetId = req.params.datasetId
  graphql(schema, DRAFT_FILES, null, null, { datasetId })
    .then(({ data }) => {
      res.send({ files: data.dataset.draft.files, datasetId })
    })
    .catch(err => {
      res.status(500)
      res.send(err)
    })
}

export const snapshotDownload = (req, res) => {
  const datasetId = req.params.datasetId
  const tag = req.params.snapshotId
  graphql(schema, SNAPSHOT_FILES, null, null, { datasetId, tag })
    .then(({ data }) => {
      res.send({ files: data.snapshot.files, datasetId, tag })
    })
    .catch(err => {
      res.status(500)
      res.send(err)
    })
}
