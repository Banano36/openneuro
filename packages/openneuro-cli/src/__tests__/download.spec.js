import { checkDestination } from '../download.js'

jest.mock('../config.js')

let errorSpy
let dirSpy

beforeEach(() => {
  errorSpy = jest.spyOn(console, 'error').mockImplementation()
  dirSpy = jest.spyOn(console, 'dir').mockImplementation()
})
afterEach(() => {
  errorSpy.mockRestore()
  dirSpy.mockRestore()
})

describe('download.js', () => {
  describe('checkDestination()', () => {
    it('throws an error on existing directories', () => {
      expect(() =>
        checkDestination('package.json'),
      ).toThrowErrorMatchingSnapshot()
    })
  })
})
