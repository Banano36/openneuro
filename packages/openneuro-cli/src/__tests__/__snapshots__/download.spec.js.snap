// Vitest Snapshot v1

exports[`download.js > checkDestination() > throws an error on existing directories 1`] = `"Target file \\"package.json\\" must be a directory"`;
